concepts/&str/about.md

```diff
@@ -1,3 +1,69 @@
 # About
 
-TODO
+There are two ways that Rust implements an array of characters: [str][str] and [String][String].
+
+`str`, also known as a "string slice", is usually used with a reference as `&str`.
+
+To know why, it may help to understand why the following line of code will not compile
+
+```rust
+const my_const: str = "Hello, World!";
+
+// the size for values of type `str` cannot be known at compilation time
+// the trait `Sized` is not implemented for `str`
+```
+
+Even though the literal "Hello, World!" length can be known at compilation time, as a `str` it is a sequence of unknown size.
+But as a _reference_ to a literal (`&str`), it has a "fat pointer" which holds the address and length of the literal in memory.
+Fat pointers are of a known size, and so this line will compile
+
+```rust
+const my_const: &str = "Hello, World!";
+```
+
+A mutable string slice (`mut &str`) can have its valued replaced, but not modified in place.
```

Comment:
```suggestion
A mutable binding to a string slice can have its valued replaced, but not modified in place.
```

Two changes in the suggestion:

- Add the word "binding", because that's what's mutable, not the reference
- remove the type, because `mut &str` isn't really a thing you'll see in rust.